### ISSUE TITLE
An operation is taking longer than expected

### ISSUE THREAD

**Issue by beaufortfrancois** [yoichiro/chromeos-filesystem-dropbox#4]
After mounting successfully the app and listing all my files at the root, I see this error message.

![screenshot 2015-02-09 at 7 44 12 pm](https://cloud.githubusercontent.com/assets/634478/6112973/298d4a24-b094-11e4-80e6-afc626bcb8b5.png)

**Comment by yoichiro** [comment]
@beaufortfrancois How many files does your root directory have? And, could you capture the console output in the DevTools at the time?

**Comment by beaufortfrancois** [comment]
I just figured out it occurs when Background page comes inactive and I launch the files.app again. Here's my JS console log:

```
onGetMetadataRequested: thumbnail=false
dropbox_fs.js:89 Object
extensions::uncaught_exception_handler:8 Error in event handler for fileSystemProvider.onGetMetadataRequested: TypeError: Cannot read property 'getMetadata' of null
    at DropboxFS.onGetMetadataRequested (chrome-extension://innelajgegmmacdlfnbceflcodcimonk/dropbox_fs.js:90:25)
    at null.<anonymous> (chrome-extension://innelajgegmmacdlfnbceflcodcimonk/dropbox_fs.js:197:14)
dropbox_fs.js:88 onGetMetadataRequested: thumbnail=false
dropbox_fs.js:89 Object
extensions::uncaught_exception_handler:8 Error in event handler for fileSystemProvider.onGetMetadataRequested: TypeError: Cannot read property 'getMetadata' of null
    at DropboxFS.onGetMetadataRequested (chrome-extension://innelajgegmmacdlfnbceflcodcimonk/dropbox_fs.js:90:25)
    at null.<anonymous> (chrome-extension://innelajgegmmacdlfnbceflcodcimonk/dropbox_fs.js:197:14)
background.js:16 Resumed
dropbox_fs.js:88 onGetMetadataRequested: thumbnail=false
dropbox_fs.js:89 Object
dropbox_fs.js:88 onGetMetadataRequested: thumbnail=false
dropbox_fs.js:89 Object
dropbox_fs.js:88 onGetMetadataRequested: thumbnail=false
dropbox_fs.js:89 Object
```

Will go deeper and let you know.

**Comment by beaufortfrancois** [comment]
OK. So here's a simple fix that is obviously not enough:

```javascript
  DropboxFS.prototype.onGetMetadataRequested = function(options, successCallback, errorCallback) {
    console.log("onGetMetadataRequested: thumbnail=" + options.thumbnail);
    console.log(options);

    // Resume first!
    this.resume(function() {

      this.dropbox_client_.getMetadata(
        options.entryPath, options.thumbnail, function(entryMetadata) {
          successCallback(entryMetadata);
        }.bind(this), errorCallback);
    }.bind(this));
  };
```

**Comment by yoichiro** [comment]
@beaufortfrancois Hmm... Yes, your code (add the resume() call) will work, but I guess that this way is not good. Well, I think that this root reason is that onGetMetadataRequested event occurs before completing of the resuming process. That is, each event listener for FSP should be registered after completing of mounting or resuming process. Currently, when the background context is loaded, then the event handlers are registered. I guess that this timing is bad.

**Comment by yoichiro** [comment]
@beaufortfrancois I could not reproduce this issue... Do you have steps to reproduce this issue completely?

**Comment by beaufortfrancois** [comment]
Let me know if I can help furthermore.

**Comment by yoichiro** [comment]
Sorry, I have a fever (39.1 degree)...

**Comment by beaufortfrancois** [comment]
![image](https://cloud.githubusercontent.com/assets/634478/6147670/3c684eac-b1fa-11e4-84c3-b6ccf9077d4f.png)

**Comment by yoichiro** [comment]
@beaufortfrancois Haha, thanks. I thought a good solution to fix this issue, as the result, I decided that your idea is better. I just created the pull request, and it will be maged, if you confirm it. https://github.com/yoichiro/chromeos-filesystem-dropbox/pull/6

**Comment by yoichiro** [comment]
Sorry, I just merged #6 diff to the master branch. If you find any issue, feel free to let me know that.

**Comment by beaufortfrancois** [comment]
Just rebased and this is working great! Nice work @yoichiro!
